docker/indexer/storage/storage.go

```diff
@@ -80,26 +112,40 @@ func New(ctx context.Context, projectID string) (*Store, error) {
 
 // Exists checks whether a name/hash pair already exists in datastore.
 func (s *Store) Exists(ctx context.Context, addr string, hashType string, hash plumbing.Hash) (bool, error) {
-	if _, ok := s.cache.Load(fmt.Sprintf(keyFmt, addr, hashType, hash)); ok {
+	if _, ok := s.cache.Load(fmt.Sprintf(docKeyFmt, addr, hashType, hash)); ok {
 		return true, nil
 	}
-	key := datastore.NameKey(kind, fmt.Sprintf(keyFmt, addr, hashType, hash), nil)
+	key := datastore.NameKey(docKind, fmt.Sprintf(docKeyFmt, addr, hashType, hash), nil)
 	tmp := &document{}
 	if err := s.dsCl.Get(ctx, key, tmp); err != nil {
 		if err == datastore.ErrNoSuchEntity {
 			return false, nil
 		}
 		return false, err
 	}
-	s.cache.Store(fmt.Sprintf(keyFmt, addr, hashType, hash), true)
+	s.cache.Store(fmt.Sprintf(docKeyFmt, addr, hashType, hash), true)
 	return true, nil
 }
 
 // Store stores a new entry in datastore.
-func (s *Store) Store(ctx context.Context, repoInfo *preparation.Result, hashType string, fileResults []processing.FileResult) error {
-	key := datastore.NameKey(kind, fmt.Sprintf(keyFmt, repoInfo.Addr, hashType, repoInfo.Commit[:]), nil)
-	_, err := s.dsCl.Put(ctx, key, newDoc(repoInfo, hashType, fileResults))
-	return err
+func (s *Store) Store(ctx context.Context, repoInfo *preparation.Result, hashType string, fileResults []*processing.FileResult) error {
+	docKey := datastore.NameKey(docKind, fmt.Sprintf(docKeyFmt, repoInfo.Addr, hashType, repoInfo.Commit[:]), nil)
```

Comment:
Can we wrap this all into a Transaction ?

Reply:
Done.